### PR TITLE
fix: allow resetting fullscreen client to nullptr

### PR DESCRIPTION
**Pull request by yury-s**
To avoid crashes like https://bugs.webkit.org/show_bug.cgi?id=264360 we need to reset wpe backend clients (see [this PR](https://github.com/WebKit/WebKit/pull/20123)) and the assert in `wpe_view_backend_set_fullscreen_client`did not allow that.

### CODE REVIEW

File: src/view-backend.c
Comment (mcatanzaro) [inline src/view-backend.c:106]: I would either remove the assert, or else parallel it in set_backend_client() and set_input_client(). Don't see any reason the fullscreen client should be different from the other two.

Comment (yury-s) [reply src/view-backend.c]: Ok, removed the assert.

**Comment by mcatanzaro** [comment]
Can you squash your changes into one commit please?

**Comment by mcatanzaro** [comment]
Looks good now, but I'm not a maintainer so can't merge.

File: src/view-backend.c
Comment (aperezdc) [inline src/view-backend.c:107]: Something that is an issue with all the `*_client` pointers we store in the `wpe_view_backend`: we do not have a way of telling client code when the clients are not needed anymore. A good idea would be to add a `destroy` callback, but of course that is out of the scope of this patch.

Comment (aperezdc) [reply src/view-backend.c:107]: @yury-s I agree with the change, but could you please add `(nullable)` annotations to the parameters in the documentation comment in `include/wpe/view-backend.h`, lines 106-107, before merging this? They need to be `(transfer none) (nullable)`.

**Review by aperezdc — CHANGES_REQUESTED** [review]
@yury-s I agree with allowing setting the fullscreen client to a null value, it makes sense. Could you take a look at the comment about changing the documentation annotations before merging? Thanks!

**Review by aperezdc — APPROVED** [review]
@yury-s Great, thanks for the documentation fixup!

**Pull request merged**
